### PR TITLE
feat: Feature/science locale

### LINKED ISSUE #899: add locale to science module
ref book module

### PR DESCRIPTION
**Pull request by SgtApone117**
fixes #899 

- added locale to science module
- made changes to unit test to handle param
- Currently only has en-us as default.

### CODE REVIEW

File: include/faker-cxx/science.h [hunk 1]
```diff
@@ -98,5 +98,5 @@ FAKER_CXX_EXPORT Unit tempUnit();
  * current.
  * @endcode
  */
-FAKER_CXX_EXPORT Unit currentUnit();
+FAKER_CXX_EXPORT Unit currentUnit(Locale locale = Locale::en_US);
```
Comment (cieslarmichal) [inline include/faker-cxx/science.h:101]: please add doxygen comments aboout new param

**Review by cieslarmichal — CHANGES_REQUESTED** [review]

**Comment by cieslarmichal** [comment]
and format code with clang format

**Comment by SgtApone117** [comment]
@cieslarmichal fixed the format using clang

File: include/faker-cxx/science.h (hunk 1 above)
Comment (SgtApone117) [reply include/faker-cxx/science.h:102]: Yes i will add it

Comment (SgtApone117) [reply include/faker-cxx/science.h:116]: I have added the comments

**Review by cieslarmichal — APPROVED** [review]

**Pull request merged**